src/main/kotlin/com/noumenadigital/npl/cli/settings/LocalSettings.kt

```diff
@@ -0,0 +1,11 @@
+package com.noumenadigital.npl.cli.settings
+
+data class LocalSettings(
+    val managementUrl: String,
+    val authUrl: String,
+    val password: String? = null,
+    val username: String? = null,
+    val clientId: String? = null,
+    val clientSecret: String? = null,
+    val clear: Boolean = false,
+)
```

Comment:
Lots of issues here:

- there are like three or four different config files and places where these properties are duplicated (YamlConfig, NamedParameters, service configs, here). Ideally there should be one or two.
- nullability is a mess. they shouldn't be nullable all over the place and assigned nulls. this should be carefully controlled and only done where meaningful.

Addressing this is too much scope for me to handle right now, though, so unfortunately I think we might have to leave this as it is.